sql/item.cc

```diff
@@ -10844,6 +10845,102 @@ const char *dbug_print_item(Item *item)
     return "Couldn't fit into buffer";
 }
 
+#define DBUG_BIG_BUF_SIZE 20480
+static char big_buf[DBUG_BIG_BUF_SIZE];
+
+const char *dbug_print_items(Item *item);
+void dbug_add_print_items(Item *item);
+void dbug_add_print_items(List<Item> &list)
+{
+  List_iterator<Item> li(list);
+  Item *sub_item;
+  strcpy( big_buf + strlen( big_buf ), "Children:" );
```

Comment:
Being a static buffer I assume this isn't thread safe. Perhaps a `StringBuffer` or other dynamic structure would be better. Even in a debug build.

This would eliminate these functions generally should have a greater accounting for the length of the buffer.

I haven't looked at the other optimizer trace parts but consistency is good.

Does this still satisfy the MDEV if its under a DBUG build?

Reply:
These functions aren't meant to be called from anything other than a debugger, so yeah, not thread safe at all, just like the original. But yeah, next time i revisit them i'll use something dynamic.  I just wanted something that I could use in gdb:display.